crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -189,6 +193,20 @@ where
                 }
             }
         }
+
+        // Track each top-level import, to guide import insertions.
+        if matches!(
+            &stmt.node,
+            StmtKind::Import { .. } | StmtKind::ImportFrom { .. }
+        ) {
+            let scope_index = self.ctx.scope_id();
+            if scope_index.is_global() {
+                if self.ctx.current_stmt_parent().is_none() {
+                    self.importer.visit_import(stmt);
+                }
+            }
```

Comment:
Nit: Surprising that clippy isn't flagging this
```suggestion
            if scope_index.is_global() &&self.ctx.current_stmt_parent().is_none() {
                self.importer.visit_import(stmt);
            }
```

Reply:
Changed here.